openedx/core/djangoapps/programs/utils.py

```diff
@@ -424,3 +432,48 @@ def _attach_run_mode_upgrade_url(self, run_mode):
                 run_mode['upgrade_url'] = None
         else:
             run_mode['upgrade_url'] = None
+
+
+def deprecated_start_datetime_text(start_date, advertised_start=None):
+    """
+    Thinking of using this function? Don't!
+
+    Use the edx-ui-toolkit DateUtils instead!
+
+    https://github.com/edx/edx-ui-toolkit
+
+    This is a helper function to return a formatted date string corresponding
+    to a date, with an optional 'advertised_start' variable.
+    Calculates text to be shown to user regarding a datetime.
+
+    Prefers .advertised_start, then falls back to .start_date.
+
+    Arguments:
+        start_date (datetime): A datetime object.
+        advertised_start (string): (optional) A date or time descriptor that
+            could be a string rep of a datetime, (e.g. 1918-11-11 06:35) or
+            could be a simple descriptive string (e.g. 'Fall 1918')
+
+    Returns:
+        (string)
+        a string representation of a datetime, in short format (e.g. Nov 11th, 1918)
+        or the advertised_start variable if not parsable as a datetime
+
+
+    """
+    if advertised_start is not None:
+        try:
+            # from_json either returns a Date, returns None, or raises a ValueError
+            parsed_advertised_start = Date().from_json(advertised_start)
+            if parsed_advertised_start is not None:
+                return strftime_localized(parsed_advertised_start.astimezone(utc), "SHORT_DATE")
+        except ValueError:
+            pass
+        return advertised_start.title()
+    elif start_date != DEFAULT_ENROLLMENT_START_DATE:
+        return strftime_localized(start_date.astimezone(utc), "SHORT_DATE")
+    else:
+        _ = ugettext
```

Comment:
For your reference, don't do this. Just import as `from django.utils.translation import ugettext as _`.